packages/truffle-codec/lib/decode/abi.ts

```diff
@@ -372,3 +367,33 @@ function* decodeAbiStructByPosition(dataType: Types.StructType, location: AbiLoc
     value: decodedMembers
   };
 }
+
+//note that this function takes the start position as a *number*; it does not take a pointer
```

Comment:
why not?

Reply:
For the same reason as with structs.  So why does the struct function work that way?  Well, originally as you'll recall there was no separate function for structs, the code was simply copypasted in both the dynamic and static decoding functions.  The struct function came about by factoring that.

So why not take a pointer?  Because that would work differently in the static vs dynamic cases.  The dynamic function assumes that the pointer points to a *pointer* to the struct (or in this case tuple); the static function assumes that the pointer points to the struct (or tuple) itself.  So, this factored version just takes the start as a number.

Notionally you could have it take a pointer to the struct (or tuple) itself, but I don't like that, as it breaks the convention used elsewhere that with dynamic things you *always* use a pointer to the pointer.  You could also combine the two cases a bit more, but that would require getting rid of an optimization I have in the static case where it doesn't go back to the dispatcher but just calls the static case again directly.

(Is it worth getting rid of that optimization for the sake of clarity?  I think it's not; for one, I'd dispute the "clarity" gain as it breaks the convention I mention above, and two, even if it is worth it, it's certainly not worth the effort to change now that it's already in the code. :P )

Anyway that's why.

Reply:
alright. well, since I am clearly in over my head here, I have no grounds to argue :)